src/Conversion/ONNXToKrnl/NN/Pooling.cpp

```diff
@@ -177,141 +329,197 @@ struct ONNXMaxPoolSingleOutOpLowering : public ConversionPattern {
     //         R[n][c][r1][r2] = negative_infinity;
     //         for k1 = 0 .. KH:
     //           for k2 = 0 .. KW:
-    //             t = D[n][c][s1 * r1 + k1 * d1][s2 * r2 + k2 * d2];
+    //             t = D[n][c][s1 * r1 + k1][s2 * r2 + k2];
```

Comment:
Please update the pesudo code to reflect the algorithm of average pooling. Having a separate section explaining AveragePooling's algorithm may be desirable. I wouldn't worry about duplication of documentation.